lib/cookie.js

```diff
@@ -486,6 +505,8 @@ function parse(str, strict) {
 
   // ensure a default date for sorting:
   c.creation = new Date();
+  //NOTE: high-resolution time for the sorting in runtime
+  c.hrCreation = hrTimeNow();
```

Comment:
Previous implementation caused flickering of the IETF's ordering test. Cookies were added too fast and `Date()` resolution was not enough to provide the correct order. We can reset `hrCreation` if `creation` was set manually.